apps/wallet/src/ui/app/components/ledger/ConnectLedgerModalContainer.tsx

```diff
@@ -0,0 +1,45 @@
+import { LockedDeviceError } from '@ledgerhq/errors';
+import toast from 'react-hot-toast';
+import { useNavigate } from 'react-router-dom';
+
+import { useNextMenuUrl } from '../menu/hooks';
+import { ConnectLedgerModal } from './ConnectLedgerModal';
+import {
+    LedgerConnectionFailedError,
+    LedgerNoTransportMechanismError,
+} from './LedgerExceptions';
+
+export function ConnectLedgerModalContainer() {
+    const navigate = useNavigate();
+    const accountsUrl = useNextMenuUrl(true, '/accounts');
+    const importLedgerAccountsUrl = useNextMenuUrl(
+        true,
+        '/import-ledger-accounts'
+    );
+
+    return (
+        <ConnectLedgerModal
+            onClose={() => {
+                navigate(accountsUrl);
+            }}
+            onError={(error) => {
+                navigate(accountsUrl);
+                toast.error(getLedgerErrorMessage(error));
+            }}
+            onConfirm={() => {
+                navigate(importLedgerAccountsUrl);
+            }}
+        />
+    );
+}
+
+function getLedgerErrorMessage(error: unknown) {
```

Comment:
Aren't these our errors? Can we just make the `message` on it human readable?

Reply:
I was going to re-work this in https://mysten.atlassian.net/browse/APPS-642.

Some of the errors are from the Ledger library and some are errors that we've intercepted and converted to our own errors. Right now, we're swallowing the internal exceptions which aren't good and hard for debugging so this was one thing I was going to address. I guess to directly answer your question, I didn't want to couple internal error messages to user messaging; I was thinking the errors we surface would have messages like:

```
LedgerConnectionFailedError(`Failed to connect to the user's Ledger device: ${internalErrorMessageFromLedger}`)
UI message -> Ledger connection failed. Try again.
```

Reply:
Going to go ahead and land but feel free to reply on this one :)